showcase/src/app/shared/effects/stark-error-handler.effects.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, Injector } from "@angular/core";
+import { Actions, Effect, ofType } from "@ngrx/effects";
+import { Observable } from "rxjs";
+import { map, take } from "rxjs/operators";
+
+import { STARK_TOAST_NOTIFICATION_SERVICE, StarkMessageType, StarkToastNotificationService } from "@nationalbankbelgium/stark-ui";
+import { StarkErrorHandlerActionTypes, StarkUnhandledError } from "@nationalbankbelgium/stark-core";
+
+/**
+ * This class is used to determine what to do with an error
+ */
+@Injectable()
+export class StarkErrorHandlerEffects {
+	private _starkToastNotificationService: StarkToastNotificationService;
+
+	/**
+	 * Class constructor
+	 * @param actions$ - the action to perform
+	 * @param injector - the injector of the class
+	 */
+	public constructor(private actions$: Actions, private injector: Injector) {}
+
+	@Effect({ dispatch: false })
+	public setToastMessage$(): Observable<void> {
+		return this.actions$.pipe(
+			ofType<StarkUnhandledError>(StarkErrorHandlerActionTypes.UNHANDLED_ERROR),
+			map((action: StarkUnhandledError) => {
+				this.toastNotificationService
+					.show({
+						id: "1",
+						type: StarkMessageType.ERROR,
+						key: action.error.toString(),
+						code: "abc"
+					})
+					.pipe(take(1))
+					.subscribe();
+			})
+		);
+	}
+
+	/**
+	 * Gets the StarkToastNotificationService from the Injector (this is tried only once).
+	 * It returns 'undefined' if the service is not found (the ToastNotification module is not imported in the app).
+	 * @returns StarkToastNotificationService
+	 */
```

Comment:
In this case, this JSDoc is not accurate since the use case is a bit different, I suggest putting the following instead:

```
Gets the StarkToastNotificationService from the Injector.
@throws When the service is not found (the ToastNotification module is not imported in the app)
```